server/src/main/java/org/apache/druid/initialization/ServerInjectorBuilder.java

```diff
@@ -118,6 +135,37 @@ public Injector build()
     return new ExtensionInjectorBuilder(serviceBuilder).build();
   }
 
+  /**
+   * Rough bridge solution for Hadoop indexing that needs server-like Injector but can't run jetty 12
+   */
+  public Injector buildWithoutJettyModules()
```

Comment:
Let's try to common out the parts of this method with the existing `build` method.